pallets/pool-system/src/impls.rs

```diff
@@ -154,24 +154,6 @@ impl<T: Config> PoolMutate<T::AccountId, T::PoolId> for Pallet<T> {
 			T::MaxTranches,
 		>::from_input::<T::StringLimit>(pool_id, tranche_inputs.clone(), now)?;
 
-		for (tranche, tranche_input) in tranches.tranches.iter().zip(&tranche_inputs) {
-			let token_name = tranche_input.metadata.token_name.clone();
-			let token_symbol = tranche_input.metadata.token_symbol.clone();
-
-			// The decimals of the tranche token need to match the decimals of the pool
-			// currency. Otherwise, we'd always need to convert investments to the decimals
-			// of tranche tokens and vice versa
-			let decimals = match T::AssetRegistry::metadata(&currency) {
-				Some(metadata) => metadata.decimals,
-				None => return Err(Error::<T>::MetadataForCurrencyNotFound.into()),
-			};
-
-			let metadata = tranche.create_asset_metadata(decimals, token_name, token_symbol);
-
-			T::AssetRegistry::register_asset(Some(tranche.currency.into()), metadata)
```

Comment:
Why is it no longer registered?

Reply:
It is! This was just moved down a few lines (L208+).

Reply:
The reason why we want to move this closure is to change the dispatch order of the events: Pool creation should be first and then the tranche token registration events.